web3.go

```diff
@@ -467,6 +467,9 @@ func ConvertArgument(abiType abi.Type, param interface{}) (interface{}, error) {
 		s = strings.TrimPrefix(s, "[")
 		s = strings.TrimSuffix(s, "]")
 		inputArray := strings.Split(s, ",")
+		if s == "" {
+			inputArray = nil
```

Comment:
Would this be better as an empty array rather than nil?

Reply:
I just follow this stackoverflow answer
https://stackoverflow.com/a/16973160

> Setting the slice to nil is the best way to clear a slice. nil slices in go are perfectly well behaved and setting the slice to nil will release the underlying memory to the garbage collector.

And according to official document
https://go.dev/tour/moretypes/12

>   The zero value of a slice is nil.
>
> A nil slice has a length and capacity of 0 and has no underlying array.